src/julia_internal.h

```diff
@@ -108,9 +108,9 @@ static inline void jl_assume_(int cond)
 #  define JL_USE_IFUNC 0
 #endif
 
-#ifdef JULIA_ENABLE_THREADING
 JL_DLLEXPORT void jl_uv_lock(void);
 JL_DLLEXPORT void jl_uv_unlock(void);
+#ifdef JULIA_ENABLE_THREADING
```

Comment:
Just use `JL_LOCK` instead of the wrapper, it takes care of everything.

Reply:
JL_LOCK is not empty, it adds GC-related stuff even for non-threaded code